document-portal/document-portal.c

```diff
@@ -412,8 +412,6 @@ validate_fd_common (int fd,
   if (fd == -1 ||
       /* Must be able to get fd flags */
       (fd_flags = fcntl (fd, F_GETFL)) == -1 ||
-      /* Must be O_PATH */
-      ((fd_flags & O_PATH) != O_PATH) ||
       /* Must not be O_NOFOLLOW (because we want the target file) */
       ((fd_flags & O_NOFOLLOW) == O_NOFOLLOW) ||
```

Comment:
The O_NOFOLLOW check is only really relevant to O_PATH descriptors, but I is probably harmless for other paths.

It looks like the O_NOFOLLOW flag is reported by fcntl(..., F_GETFL) for regular file descriptors opened that way, so the question is whether there are legitimate cases where the client would want to use the flag.

Reply:
The only use for it would be if you only want to export a file if it is not a symlink, which seems not very useful. But I agree, that the check is useless for non-O_PATH, so might as well not keep it.